src/main/scala/com/gd/twitteranalytics/ReportProcessor.scala

```diff
@@ -0,0 +1,50 @@
+package com.gd.twitteranalytics
+
+import org.apache.spark.sql.expressions.Window
+import org.apache.spark.sql.expressions.scalalang.typed.{count => typedCount}
+import org.apache.spark.sql.functions.{count, current_date, lit, row_number}
+import org.apache.spark.sql.{DataFrame, Dataset, Encoders, SparkSession}
+
+object ReportProcessor {
```

Comment:
two points:
1) I see that for some reason you call objects and classes "files":
> ... into a new file

To my mind, a "file" is not quite useful abstraction when we talk about OOP/FP design and implementaiton.

2) 
> ... methods  have been moved here from TwitterReportGenerator

I don't see TwitterReportGenerator changed. this is not what "moved" implies.